native/dgesvd.go

```diff
@@ -0,0 +1,1241 @@
+			} else if wantuo && wantvn {
+				// Path 2
+				wrkbl = n + lwork_dgeqrf
+				wrkbl = max(wrkbl, n+lwork_dorgqr_n)
+				wrkbl = max(wrkbl, 3*n+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = max(n*n+wrkbl, n*n+m*n+n)
+			} else if wantuo && wantvs {
+				// Path 3
+				wrkbl = n + lwork_dgeqrf
+				wrkbl = max(wrkbl, n+lwork_dorgqr_n)
+				wrkbl = max(wrkbl, 3*n+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = max(n*n+wrkbl, n*n+m*n+n)
+			} else if wantus && wantvn {
+				// Path 4
+				wrkbl = n + lwork_dgeqrf
+				wrkbl = max(wrkbl, n+lwork_dorgqr_n)
+				wrkbl = max(wrkbl, 3*n+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = n*n + wrkbl
+			} else if wantus && wantvo {
+				// Path 5
+				wrkbl = n + lwork_dgeqrf
+				wrkbl = max(wrkbl, n+lwork_dorgqr_n)
+				wrkbl = max(wrkbl, 3*n+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = 2*n*n + wrkbl
+			} else if wantus && wantvas {
+				// Path 6
+				wrkbl = n + lwork_dgeqrf
+				wrkbl = max(wrkbl, n+lwork_dorgqr_n)
+				wrkbl = max(wrkbl, 3*n+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = n*n + wrkbl
+			} else if wantua && wantvn {
+				// Path 7
+				wrkbl = n + lwork_dgeqrf
+				wrkbl = max(wrkbl, n+lwork_dorgqr_m)
+				wrkbl = max(wrkbl, 3*n+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = n*n + wrkbl
+			} else if wantua && wantvo {
+				// Path 8
+				wrkbl = n + lwork_dgeqrf
+				wrkbl = max(wrkbl, n+lwork_dorgqr_m)
+				wrkbl = max(wrkbl, 3*n+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = 2*n*n + wrkbl
+			} else if wantua && wantvas {
+				// Path 9
+				wrkbl = n + lwork_dgeqrf
+				wrkbl = max(wrkbl, n+lwork_dorgqr_m)
+				wrkbl = max(wrkbl, 3*n+lwork_dgebrd)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_q)
+				wrkbl = max(wrkbl, 3*n+lwork_dorgbr_p)
+				wrkbl = max(wrkbl, bdspac)
+				maxwrk = n*n + wrkbl
+			}
+		} else {
+			// Path 10: m > n
+			impl.Dgebrd(m, n, a, lda, s, dum, dum, dum, dum, -1)
+			lwork_dgebrd := int(dum[0])
+			maxwrk = 3*n + lwork_dgebrd
+			if wantus || wantuo {
+				impl.Dorgbr(lapack.ApplyQ, m, n, n, a, lda, dum, dum, -1)
+				lwork_dorgbr_q = int(dum[0])
+				maxwrk = max(maxwrk, 3*n+lwork_dorgbr_q)
+			}
+			if wantua {
+				impl.Dorgbr(lapack.ApplyQ, m, m, n, a, lda, dum, dum, -1)
+				lwork_dorgbr_q := int(dum[0])
+				maxwrk = max(maxwrk, 3*n+lwork_dorgbr_q)
+			}
+			if !wantvn {
+				maxwrk = max(maxwrk, 3*n+lwork_dorgbr_p)
+			}
+			maxwrk = max(maxwrk, bdspac)
+		}
+	} else {
```

Comment:
The fortran tests `min(m, n) > 0` *here*‽ (now in correct spot - the bewilderment it engendered knocked me off mark).